internal/gitstrap/apply.go

```diff
@@ -205,3 +207,37 @@ func (g *Gitstrap) editTeamBySlug(m *spec.Model) error {
 	m.Metadata.FromGithubTeam(gT)
 	return nil
 }
+
+func (g *Gitstrap) applyProtection(m *spec.Model) error {
+	ctx, cancel := g.newContext()
+	defer cancel()
+	owner := g.getOwner(m)
+	name, err := getSpecifiedName(m)
+	if err != nil {
+		return err
```

Comment:
Let's return a bit more information about the error and wrap the "cause" using `%w` format parameter, e.g. `fmt.Errorf("protection name is required: %w", err)`